node/src/components/small_network/tasks.rs

```diff
@@ -134,11 +141,16 @@ where
 
     // Setup connection id and framed transport.
     let connection_id = ConnectionId::from_connection(transport.ssl(), context.our_id, peer_id);
-    let framed = framed_transport(transport, context.chain_info.maximum_net_message_size);
+    let framed_transport = framed_transport(transport, context.chain_info.maximum_net_message_size);
 
     // Negotiate the handshake, concluding the incoming connection process.
-    match negotiate_handshake::<P, _>(&context, framed, connection_id).await {
-        Ok((framed, public_addr, peer_consensus_public_key, is_joiner)) => {
+    match negotiate_handshake::<P, _>(&context, framed_transport, connection_id).await {
+        Ok(HandshakeOutcome {
+            framed_transport,
+            public_addr,
+            peer_consensus_public_key,
+            is_peer_joiner: is_joiner,
```

Comment:
We could do without the rename here, for clarity, I think. I know it keeps the diff smaller, but I'd be too tempted to remove the alias later on :)